examples/mxnet/gcn/gcn_spmv.py

```diff
@@ -0,0 +1,187 @@
+        super(NodeUpdateModule, self).__init__()
+        self.linear = gluon.nn.Dense(out_feats, activation=activation)
+        self.dropout = dropout
+        self.normalization = normalization
+
+    def forward(self, node):
+        if self.normalization == 'sym':
+            h = node.data['h'] * node.data['out_degree'].sqrt()
+        else:
+            h = node.data['h']
+        if self.normalization == 'sym':
+            accum = node.data['accum'] / node.data['in_degree'].sqrt()
+        elif self.normalization == 'left':
+            accum = node.data['accum'] / node.data['in_degree']
+        else:
+            accum = node.data['accum']
+        accum = self.linear(accum)
+        if self.dropout:
+            accum = mx.nd.Dropout(accum, p=self.dropout)
+        return {'h': mx.nd.concat(h, accum, dim=1)}
+
+
+class GCN(gluon.Block):
+    def __init__(self,
+                 g,
+                 in_feats,
+                 n_hidden,
+                 n_classes,
+                 n_layers,
+                 activation,
+                 dropout,
+                 normalization,
+                 ):
+        super(GCN, self).__init__()
+        self.g = g
+        self.dropout = dropout
+        self.normalization = normalization
+
+        self.inp_layer = gluon.nn.Dense(n_hidden, activation)
+
+        self.conv_layers = gluon.nn.Sequential()
+        for i in range(n_layers):
+            self.conv_layers.add(NodeUpdateModule(n_hidden, activation, dropout, normalization))
+
+        self.out_layer = gluon.nn.Dense(n_classes)
+
+
+    def forward(self, features):
+        emb_inp = [features, self.inp_layer(features)]
+        if self.dropout:
+            emb_inp[-1] = mx.nd.Dropout(emb_inp[-1], p=self.dropout)
+
+        self.g.ndata['h'] = mx.nd.concat(*emb_inp, dim=1)
+        for layer in self.conv_layers:
+            if self.normalization == 'sym':
+                self.g.ndata['h'] = self.g.ndata['h'] / self.g.ndata['out_degree'].sqrt()
+            self.g.update_all(fn.copy_src(src='h', out='m'),
+                              fn.sum(msg='m', out='accum'),
+                              layer)
+
+        emb_out = self.g.ndata.pop('h')
+        return self.out_layer(emb_out)
+
+
+def main(args):
+    # load and preprocess dataset
+    data = load_data(args)
+
+    if args.self_loop:
+        data.graph.add_edges_from([(i,i) for i in range(len(data.graph))])
+
+    features = mx.nd.array(data.features)
+    labels = mx.nd.array(data.labels)
+    mask = mx.nd.array(data.train_mask)
+    in_degree  = mx.nd.array([data.graph.in_degree(i)
+                              for i in range(len(data.graph))])
+    out_degree = mx.nd.array([data.graph.out_degree(i)
+                              for i in range(len(data.graph))])
+
+    in_feats = features.shape[1]
+    n_classes = data.num_labels
+    n_edges = data.graph.number_of_edges()
+
+    if args.gpu <= 0:
+        cuda = False
```

Comment:
please remove this. it's my fault. When i ported the GCN impl from Pytorch, I forgot to remove this line.